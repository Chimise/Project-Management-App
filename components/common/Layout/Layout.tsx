import React, { Fragment } from "react";
import { AnimatePresence } from "framer-motion";
import Toast from "../../ui/Toast";
import useUI from "../../../hooks/useUI";

interface LayoutProps {
  children: React.ReactNode;
}

const Layout = ({ children }: LayoutProps) => {
  const { toastIsVisible, status, message, closeToastHandler } = useUI();
  return (
    <Fragment>
      <Toast
        isVisible={toastIsVisible}
        status={status}
        message={message}
        onClose={closeToastHandler}
      />
      <AnimatePresence mode="wait">{children}</AnimatePresence>
    </Fragment>
  );
};

export default Layout;
