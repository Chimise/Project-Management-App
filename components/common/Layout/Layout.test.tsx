import "@testing-library/jest-dom";
import {
  render,
} from "../../../test/test-utils";
import router, {
  mockUseRouter,
} from "../../../test/router";
import Layout from './Layout';

beforeEach(() => {
  jest.useFakeTimers();
  jest.clearAllMocks();
});

afterEach(() => {
  jest.useRealTimers();
});

mockUseRouter();



test('It renders unchanged', () => {
    const {container} = render(<Layout>Hello</Layout>)
    expect(container.firstChild).toMatchSnapshot();
})
