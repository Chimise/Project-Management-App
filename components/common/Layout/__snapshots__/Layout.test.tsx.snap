// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`It renders unchanged 1`] = `null`;
